### PR TITLE
[BEAM-4524] Use SHA256 for artifact validation

### PR DESCRIPTION
**Pull request by angoenka**
Making go changes required to support sha256 for artifact validation.

Continuation work from https://github.com/apache/beam/pull/6583
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by angoenka** [comment]
cc: @lostluck

**Comment by angoenka** [comment]
Run Python Flink ValidatesRunner

**Comment by angoenka** [comment]
Run Go PostCommit

**Comment by angoenka** [comment]
Run Python Flink ValidatesRunner

**Comment by angoenka** [comment]
Run Go PostCommit

**Comment by angoenka** [comment]
Run Python Flink ValidatesRunner

**Comment by lostluck** [comment]
LGTM assuming the Go PostCommit passes.

File: sdks/go/pkg/beam/artifact/gcsproxy/staging.go [hunk 1]
```diff
@@ -18,8 +18,9 @@ package gcsproxy
 
 import (
 	"bytes"
-	"crypto/md5"
+	"crypto/sha256"
 	"encoding/base64"
```
Comment (robertwb) [inline sdks/go/pkg/beam/artifact/gcsproxy/staging.go]: Do we still need this?

Comment (angoenka) [reply sdks/go/pkg/beam/artifact/gcsproxy/staging.go]: No, we don't need it. 
I must have missed removing it.

**Comment by angoenka** [comment]
Run Python Flink ValidatesRunner

**Comment by angoenka** [comment]
Run Go PostCommit

**Comment by angoenka** [comment]
Going ahead with the merge as tests are passing.

**Comment by swegner** [comment]
FYI, this appears to break Java post-commits: https://issues.apache.org/jira/browse/BEAM-6080

**Comment by swegner** [comment]
FYI, this regression caused Dataflow test jobs to hang indefinitely; I went into the GCP project and manually cancelled a bunch of jobs which were crash-looping on [this error](https://issues.apache.org/jira/browse/BEAM-6080) for nearly 24 hours.

It seems like a small bug like this shouldn't be able to hang jobs indefinitely and eat up our GCP quotas. Are we missing some check to prevent indefinite crash-looping?

**Comment by swegner** [comment]
This PR was rolled back in PR #7067 due to java post-commit regression according to our [rollback-first policy](https://beam.apache.org/contribute/postcommits-policies-details/index.html#rollback_first). 

You can roll-forward by reverting the rollback commit (`git revert 1b241f9517342c73ed2f0a73251858ee67c7e191`), and then build your fix on top of it. Be sure to run Java post-commit tests on your fix ("Run Java PostCommit")

For more information, see the post-commit test guide, ["My change was rolled back due to a test failure"](https://beam.apache.org/contribute/postcommits-policies/#pr-rolled-back).

**Pull request merged**